.github/workflows/docker-image.yml

```diff
@@ -7,25 +7,35 @@ on:
     branches: [ "master" ]
 
 jobs:
-
   build:
-
     runs-on: ubuntu-latest
-
     steps:
-    - uses: actions/checkout@v3
-    - name: Build the Docker image
-      run: docker build . --file Dockerfile --tag ghcr.io/havenweb/haven:$(echo $GITHUB_SHA | head -c7) --tag ghcr.io/havenweb/haven:latest
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Get tag
+        id: get_tag
+        run: echo "tagname=$(echo $GITHUB_SHA | head -c7)" >> $GITHUB_OUTPUT
+        
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx 
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to GitHub Container Registry 
+        uses: docker/login-action@v3 
+        with:
+          registry: ghcr.io 
+          username: ${{ github.repository_owner }} 
+          password: ${{ secrets.GITHUB_TOKEN }} 
+
+      - name: Build and push image 
+        uses: docker/build-push-action@v6 
+        with: 
+          context: . 
+          platforms: linux/amd64, linux/arm64 
+          push: true 
```

Comment:
Could we add a condition here?

`push: github.ref == 'refs/heads/master'`

or similar?

Reply:
Huh that may actually work

Reply:
Let's try